iotc-generic-c-sdk/paho-c-impl/src/iotc_paho_client.c

```diff
@@ -30,6 +30,9 @@ static void paho_deinit(void) {
 }
 
 static int on_c2d_message(void *context, char *topicName, int topicLen, MQTTClient_message *message) {
+    (void) context;
```

Comment:
We should not really need these unused parameter "fixes". These are callbacks and it's quite usual that one does not use all parameters of a callback. Default for gcc is Wno-unused-parameters. These types of warnings are not reported even with the Wall flag. I mean who writes (void)argv; (void)argc;...If we were to support every pedantic compiler option there is, our code would double in size :)

Anyways.... I am NOT against it. Either is fine in my opinion as long as we don't go overboard on this in the future. I also **do support** the qos explicit (void) in the changes above to indicate that in that we are not able to control QOS with the Azure client.